vector/src/main/java/im/vector/app/features/matrixto/MatrixToBottomSheetViewModel.kt

```diff
@@ -296,8 +297,13 @@ class MatrixToBottomSheetViewModel @AssistedInject constructor(
         }
         viewModelScope.launch {
             try {
-                session.joinRoom(action.roomId, null, action.viaServers?.take(3) ?: emptyList())
-                _viewEvents.post(MatrixToViewEvents.NavigateToRoom(action.roomId))
+                session.joinRoom(action.roomIdOrAlias, null, action.viaServers?.take(3) ?: emptyList())
+                val roomId: String = if (MatrixPatterns.isRoomAlias(action.roomIdOrAlias)) {
```

Comment:
do you think it's worth extracting out a method for converting `roomIdOrAlias` to just `roomId`?